js/modules/k6/experimental/tracing/trace_id.go

```diff
@@ -76,7 +94,7 @@ func (t TraceID) Encode() (string, error) {
 	// of available bytes left after writing the prefix, code and timestamp (index n)
 	// is filled with random bytes.
 	randomness := make([]byte, 16-n)
-	err := binary.Read(rand.Reader, binary.BigEndian, randomness)
+	err := binary.Read(t.randSource, binary.BigEndian, randomness)
```

Comment:
This is more of a general comment. But it seems strange that calling `TraceID.Encode` multiple times will get you *different* values. 

Like it seems to me this should just basically be a single function instead of having code that, creates a struct and then calls `Encode` on it as the only thing it does.

Reply:
I've experimented around it. I have an idea of how to decouple trace id initialisation from encoding, but because the random part doesn't have a fixed size, it's gonna make things a bit more involved than I anticipated.

I'm already working on improving this in a separate PR. Resolving this for now, and stay tuned for the upcoming refactor PR 👍